main.rb

```diff
@@ -0,0 +1,84 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    counter = 0
+    while counter < size
+      yield(self[counter])
+      counter += 1
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    while i < size
+      yield(self[i], i)
+      i += 1
+    end
+  end
+
+  def my_select
+    output = []
+    my_each { |ele| output << ele if yield(ele) }
+    output
+  end
+
+  def my_all?
+    output = true
+    my_each { |ele| output = false if yield(ele) != true } if block_given?
+    my_each { |ele| output = false if ele.nil? || ele == false } unless block_given?
+    output
+  end
+
+  def my_any?
+    output = false
+    my_each do |ele|
+      output = true if yield(ele) == true
+    end
+    output
+  end
+
+  def my_none?(arg = nil)
+    output = true
+    if !block_given?
+      if arg.nil?
+        my_each { |ele| output = false if ele }
+      else
+        my_each { |ele| output = false if ele == arg }
+      end
+    else
+      my_each { |ele| output = false if yield(ele, arg) == true }
```

Comment:
only this line is enough for the my_none functionality. Set output = true 
The set output to false if the yield is true
return output. 
The argument is not needed